src/service/setupService.ts

```diff
@@ -159,44 +159,149 @@ export class SetupService {
+                                        TelemetryEventName.SETUP_DOWNLOAD_PYTHON
+                                    );
+                                    open(CONSTANTS.LINKS.DOWNLOAD_PYTHON);
+                                };
+                                showPrivacyModal(
+                                    okAction,
+                                    CONSTANTS.INFO.THIRD_PARTY_WEBSITE_PYTHON
+                                );
+                            }
+                        });
+                    // no python installed, cannot get path
+                    return "";
+                } else {
+                    // "python" didn't resolve to anything, trying "python3"
+                    return this.getCurrentPythonExecutablePath(true);
+                }
+            }
+            if (
+                !(await this.validatePythonVersion(
+                    originalpythonExecutablePath
+                ))
+            ) {
+                this.telemetryAI.trackFeatureUsage(
+                    TelemetryEventName.SETUP_INVALID_PYTHON_VER
+                );
+                if (tryingPython3) {
+                    // if we're trying python3, it means we already tried python and it
+                    // all doesn't seem to work, but it got this far, so it means that
+                    // their system python3 version is still not above 3.7, but they
+                    // don't have a path selected.
+                    vscode.window
+                        .showInformationMessage(
+                            CONSTANTS.ERROR.INVALID_PYTHON_PATH,
+                            DialogResponses.INSTALL_PYTHON
+                        )
+                        .then(
+                            (installChoice: vscode.MessageItem | undefined) => {
+                                if (
+                                    installChoice ===
+                                    DialogResponses.INSTALL_PYTHON
+                                ) {
+                                    const okAction = () => {
+                                        open(CONSTANTS.LINKS.DOWNLOAD_PYTHON);
+                                    };
+                                    showPrivacyModal(
+                                        okAction,
+                                        CONSTANTS.INFO
+                                            .THIRD_PARTY_WEBSITE_PYTHON
+                                    );
+                                }
+                            }
+                        );
+                    return "";
+                } else {
+                    // otherwise, we ran the "python" system variable
+                    // and we can try python3
+                    return this.getCurrentPythonExecutablePath(true);
+                }
+            }
+        } else {
+            // should only be applicable if the user defined their own path
+
+            // fix path to be absolute

```

Comment:
_&ast;nit&ast;_
I feel like this block is pretty self-documenting and that the comment can be omitted.

Reply:
Might be removed in future refactoring PR, will be kept for now.